### DISCUSSION TITLE
How to get Express request headers at Prisma logging event subscription handler?

### DISCUSSION THREAD

**Discussion by cantrepro** [prisma/prisma#18034] [Q&A]
So with Prisma you can subscribe to logging events using `$on`: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#on

And I have done so like this:
```js
const logQuery = (event: Prisma.QueryEvent) => {
  console.log(`query: ${event.query}`);
};

prisma.$on('query', logQuery);
```
Now I have bunch of endpoints where controller calls Prisma and every time the query gets logged with the code mentioned above.

What I want to do is also log the request's headers in `logQuery` (the request that executed Prisma call in a controller).

Here's the controller:
```js
public endpoint = async (
  req: express.Request,
  res: express.Response
): Promise<void> => {
  // This is what I want logged in `logQuery`
  console.log(req.get('X-Some-Header'));
  // Some Prisma query that triggers `logQuery`
  await prisma.folder.create({
    data: {
      name: req.query.name,
    },
  });
  res.status(200);
};
```
How can I go about doing that without adding params to all Prisma calls?

**Comment by nurul3101** [comment]
Hi @jafarlihi 👋 

So if I am understand correctly, you want to essentially pass the request header to `logQuery` function, right?
So that every headers can be logged automatically?

**Reply by cantrepro** [reply]
Hello @nurul3101 
Yes, correct. I tried AsyncLocalStorage that gets populated in an Express middleware with the headers and then used in `logQuery` but that didn't work because the Prisma event that is emitted does not happen in the same request async context.

**Reply by nurul3101** [reply]
I am wondering if [Prisma Client Extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) could be helpful here, we have this [example](https://github.com/prisma/prisma-client-extensions/tree/main/query-logging) of Query Logging Client Extension.

**Comment by Xetoxyc** [comment]
Im curious if you found any solution to this

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
